src/Cake.Core/Tooling/ToolSettings.cs

```diff
@@ -15,6 +15,11 @@ public class ToolSettings
         public FilePath ToolPath { get; set; }
 
         /// <summary>
+        /// Gets or sets optional timeout for tool execution
+        /// </summary>
+        public int? ToolTimeout { get; set; }
```

Comment:
I buy the prefix, but why not TimeSpan? I think we should go with a TimeSpan here. The script code becomes way more cleaner when you say `tool.Timeout = TimeSpan.FromSeconds(15)` instead of `tool.TimeoutInMilliseconds = 15000`.